From 47a0644dd7755e20910da168fccac372fcafc96c Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Tue, 27 Nov 2012 16:08:44 +0200
Subject: [PORT FROM MAIN] Telephony: Fix issue in handling null alpha identifier

BZ: 60370

When alpha identifier is a provided by the SIM and
is a null data object, UE should not display
anything about the Sending but UE is showing
"Sending...".

As per 3GPP 11.14, if the alpha identifier is provided
by the SIM and is a null data object(i.e.length = '00'
and no value part), this is an indication that the ME
should not give any information to the user on the fact
that the ME is sending an Send SS/Send USSD/ Send DTMF.

Instead of returning null, retrieveAlphaId returns the
default message resulting in "Sending..." shown to the
user when it is not supposed to display anything.

This patch fixes the issue by returning null when alpha
identifier is provided and is a null data object.

Change-Id: I2f05c5383e871b4d73cb2ada39aee049787dbb28
Orig-Change-Id: Icd3aede9bcf3f81d9461095b22758f11606bc225
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/76589
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../internal/telephony/cat/ValueParser.java        |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/ValueParser.java b/src/java/com/android/internal/telephony/cat/ValueParser.java
index 6f655ed..a3454ae 100644
--- a/src/java/com/android/internal/telephony/cat/ValueParser.java
+++ b/src/java/com/android/internal/telephony/cat/ValueParser.java
@@ -285,7 +285,7 @@ abstract class ValueParser {
                     throw new ResultException(ResultCode.CMD_DATA_NOT_UNDERSTOOD);
                 }
             } else {
-                return CatService.STK_DEFAULT;
+                return null;
             }
         } else {
             return CatService.STK_DEFAULT;
-- 
1.7.4.1

